rueidiscompat/command.go

```diff
@@ -3452,3 +3462,168 @@ type TSMGetOptions struct {
 	WithLabels     bool
 	SelectedLabels []interface{}
 }
+
+type JSONSetArgs struct {
+	Key   string
+	Path  string
+	Value interface{}
+}
+
+type JSONArrIndexArgs struct {
+	Start int
+	Stop  *int
+}
+
+type JSONArrTrimArgs struct {
+	Start int
+	Stop  *int
+}
+
+type JSONCmd struct {
+	baseCmd[string]
+	expanded []any // expanded will be used at JSONCmd.Expanded
+	typ      jsonCmdTyp
+}
+
+type jsonCmdTyp int
+
+const (
+	TYP_STRING jsonCmdTyp = iota
+	TYP_ARRAY
+)
+
+// https://github.com/redis/go-redis/blob/v9.3.0/json.go#L86
+func (cmd *JSONCmd) Val() string {
+	return cmd.val
+}
+
+// https://github.com/redis/go-redis/blob/v9.3.0/json.go#L105
+func (cmd *JSONCmd) Expanded() (any, error) {
+	if cmd.typ == TYP_STRING {
+		return cmd.Val(), nil
+	}
+	// TYP_ARRAY
+	return cmd.expanded, nil
+}
+
+func (cmd *JSONCmd) Result() (string, error) {
+	return cmd.Val(), nil
+}
+
+func newJSONCmd(res rueidis.RedisResult) *JSONCmd {
+	cmd := &JSONCmd{}
+	msg, err := res.ToMessage()
+	if err != nil {
+		if err == rueidis.Nil {
+			cmd.SetVal("")
```

Comment:
why cmd.SetVal("") when err == rueidis.Nil ?

Reply:
I think I'm wrong.

There are two command using `JSONCmd` as returned value: `JSON.NUMINCRBY`, `JSON.GET`:

> `JSON.NUMINCRBY` returns a bulk string reply specified as a stringified new value for each path, or `nil`, if the matching JSON value is not a number.

> `JSON.GET` returns a bulk string representing a JSON array of string replies. Each string is the JSON serialization of each JSON value that matches a path.

According to the documentation, only `JSON.NUMINCRBY` will return `nil`, and when it's nil, we should return `rueidis.Nil` as `err`, not empty string.